client/scala/src/main/scala/ai/verta/blobs/dataset/S3.scala

```diff
@@ -76,7 +144,10 @@ object S3 {
    */
   def reduce(firstBlob: S3, secondBlob: S3): Try[S3] = {
     if (firstBlob.notConflicts(secondBlob))
-      Success(new S3(firstBlob.contents ++ secondBlob.contents))
+      Success(new S3(
+        firstBlob.contents ++ secondBlob.contents,
+        firstBlob.enableMDBVersioning && secondBlob.enableMDBVersioning
```

Comment:
Same

Reply:
Fixed + added a test that demonstrates the new expected behavior.